### PR TITLE
Make CalDAV resource sync a service and expose as occ command

### LINKED ISSUE #30294: Add command to sync calendar resources manually
Currently, a background job is syncing calendar resources. Admins should be able to trigger this sync manually via occ into the DAV app.

### PR DESCRIPTION
**Pull request by ChristophWurst**
## Summary

CalDAV resources and rooms are provided by apps. Currently the rooms and resources are read in a background job once an hour. Sometimes one wants to see the changes right away.

By moving the job logic into a reusable service we can also invoke it from an occ command.

This also drops the two unused events that aren't emitted anywhere
* ``\OCP\Calendar\Resource\ForceRefreshEvent``
* ``\OCP\Calendar\Room\ForceRefreshEvent``

## TODO

- [x] Move logic into a service
- [x] Add occ command
- [ ] Documentation

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1: makes sense

I guess we can't use the new `occ background-job:execute` instead ?

cc @come-nc

File: apps/dav/lib/Command/SyncResourcesRooms.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		$this
+			->setName('dav:sync-resources-rooms')
+			->setDescription('Sync resources and rooms');
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$this->syncService->sync();
+
+		return 0;
+	}
```
Comment (come-nc) [inline apps/dav/lib/Command/SyncResourcesRooms.php:52]: Please avoid silent commands, at least output something in verbose mode.

File: apps/dav/lib/Command/SyncResourcesRooms.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SyncResourcesRooms extends Command {
+
+	private CalendarResourcesRoomsSyncService $syncService;
+
+	public function __construct(CalendarResourcesRoomsSyncService $syncService) {
+		parent::__construct();
+		$this->syncService = $syncService;
+	}
```
Comment (come-nc) [inline apps/dav/lib/Command/SyncResourcesRooms.php:40]:
```suggestion
	public function __construct(
		private CalendarResourcesRoomsSyncService $syncService,
	) {
		parent::__construct();
	}
```
Use the force Luke (if you do not need backports)

**Review by come-nc — COMMENTED** [review]
I think using background-job:execute works too in this case, given the job is always planned.

But that requires using background-job:list first to get the id, so I have no real problem with having a specific command for it.
That said we should look into improving background-job:execute to be able to run jobs by class name, and background-job:list to be able so search for jobs without knowing the exact name.

**Pull request closed without merging**